### DISCUSSION TITLE
Question about Axios template

### DISCUSSION THREAD

**Discussion by mxmory** [acacode/swagger-typescript-api#1135] [Q&A]
Generated axios request template:
<img width="609" alt="Screenshot 2022-09-07 at 23 28 24" src="https://user-images.githubusercontent.com/11426035/188972030-36d4d65b-ad1b-43ea-b851-2ec6cdfb0e2d.png">

I have 2 different schemas for request and response:

Request schema: (**id** and **color** are optional to create this entity)
```
StageModel {
  id?: number;
  name: string;
  color?: string;
  ...
  ...
}
```

Response schema: (no optional fields because I'm sure they will be presented in response and don't want typescript to complain like `'id' is possibly undefined` in future use in my react components)
```
StageModelResponse {
  id: number;
  name: string;
  color: string;
  ...
  ...
}
```

So the question is: can I generate different types and put them to axios request and response respectively? Or maybe I'm doing this all wrong and this can be achieved in more correct/elegant way? Someone help please! Thanks in advance 🙏

**Comment by Liwoj** [comment]
imho T represents response type. Request doesn't need to be a generic type as it is passed only to a generated action method where it is transformed into `FullRequestParams`. See an example below:

```
authorizationAuthorizeClient: (cisId: string, params: RequestParams = {}) =>
      this.request<ClientAuthorizationResponseModel, any>({
        path: `/auth/client/${cisId}`,
        method: "POST",
        format: "json",
        ...params,
      }),
```
